### ISSUE TITLE
Document the expected behavior with feasibility sense

### ISSUE THREAD

**Issue by matbesancon** [jump-dev/MathOptInterface.jl#1832]
Min and Max are less problematic, but it is less clear whether feasibility should erase the current objective function

**Comment by matbesancon** [comment]
Possible behaviors:

1. setting feasibility erases the objective function, the user has to read an objective function afterwards.
Corner case: when in feasibility mode, can a user still modify the objective / add one? And in that case, if it is stored, should setting the sense as feasibility again re-erase the objective?

2. setting the feasibility sense should not erase the objective before optimizing. So setting min, then feasibility, then min should not have modified the behavior nor the objective function.
Corner case: for solvers not supporting feasibility, this likely means erasing the objective function right before optimizing. In this case, the objective function is erased at `optimize!` time.

3. Setting feasibility should never erase the objective function.
Corner case: again for solvers not supporting feasibility, this implies putting aside the objective function before optimizing, then setting it back. In many cases however, this will make the solver state not optimized
```
function MOI.optimize!(o::Optimizer)
    if feasibility
        f = objective_function(o)
        erase_objective_function!(o)
        _optimize!(o)
        # makes the solver state dirty
        set_objective(o, f)
        return
    end
    # proceed normally
end
```

**Comment by odow** [comment]
The answer is intended to be 1)

This is a bit messy, but setting `FEASIBILITY_SENSE` does clear the objective:
https://github.com/jump-dev/MathOptInterface.jl/blob/4fe8757a07749587049bfbc5de08bb29cb367cde/src/Test/test_objective.jl#L60-L84

And modifying the objective does set it:
https://github.com/jump-dev/MathOptInterface.jl/blob/4fe8757a07749587049bfbc5de08bb29cb367cde/src/Test/test_objective.jl#L513-L535

**Comment by odow** [comment]
The answer is intended to be 1)

This is a bit messy, but setting `FEASIBILITY_SENSE` does clear the objective:
https://github.com/jump-dev/MathOptInterface.jl/blob/4fe8757a07749587049bfbc5de08bb29cb367cde/src/Test/test_objective.jl#L60-L84

And modifying the objective does set it:
https://github.com/jump-dev/MathOptInterface.jl/blob/4fe8757a07749587049bfbc5de08bb29cb367cde/src/Test/test_objective.jl#L513-L535

**Comment by matbesancon** [comment]
What is supposed to happen then when setting an objective while the sense is feasible (either because set before or never set?)

**Comment by odow** [comment]
It's not really clear. You should accept the objective function and assume that they will set the objective sense later. Although I think we made a point to always set the objective sense first in the tests.

**Comment by blegat** [comment]
See https://github.com/jump-dev/MathOptInterface.jl/pull/1840#discussion_r873841439

**Comment by matbesancon** [comment]
addressed in https://github.com/jump-dev/MathOptInterface.jl/pull/1841

**Comment by odow** [comment]
Closed by #1841
